carma_wm/src/WMListenerWorker.cpp

```diff
@@ -429,6 +429,42 @@ double WMListenerWorker::getConfigSpeedLimit() const
   return config_speed_limit_;
 }
 
+std_msgs::Int32MultiArray WMListenerWorker::getIntersectionGroupIdsByCurLoc(const geometry_msgs::PoseStamped& current_pos)
+{  
+  //Get all traffic light regulatory elements along the route based on current SV position
+  lanelet::BasicPoint2d cur_loc;
+  cur_loc.x() = current_pos.pose.position.x;
+  cur_loc.y() = current_pos.pose.position.y;
+  std::vector<lanelet::CarmaTrafficLightPtr> light_v = world_model_->getLightsAlongRoute(cur_loc);
+  lanelet::CarmaTrafficLightPtr first_light = light_v.front();
+  ROS_DEBUG_STREAM("Get Value TF_LIGHT of Id: " << first_light->id());
+  
+  /**
+   * Iterate all elements in the map (The traffic light regulatory element Ids and intersection/group ids mapping). 
+   * Return the intersection/group id that map to the first traffic light regulatory element Id along the subject vehicle route
+  ***/
+  for(const auto& ligth_id_m : world_model_->traffic_light_ids_ ) 
+  {
+        std::cout << "Key:[" << ligth_id_m.first << "] Value:[" << ligth_id_m.second << "]\n";
+        if( ligth_id_m.second == first_light->id() )
+        {
+            ROS_DEBUG_STREAM("CURRENT Key:" << ligth_id_m.first << "Value:" << ligth_id_m.second);
+            ROS_DEBUG_STREAM("Get First TF_LIGHT of Id: " << first_light->id());
+            uint32_t temp = ligth_id_m.first;
+            //get intersection id and group id
+            unsigned group_id = (temp & 0xFF);
+            unsigned intersection_id = (temp >> 8);
```

Comment:
can we factor out this small method into a different method like here https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/carma_wm/include/carma_wm/WorldModelUtils.h#L116 (where we pass in the group_id and intersection_ids as a reference to be changed) 
This is just to be consistent

Reply:
addressed